benchmark/compute_accuracy.py

```diff
@@ -0,0 +1,182 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, The WGS-HGT Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+"""
+Given known HGTs, compute precision, recall and F-score for various tools
+=========================================================================
+"""
+
+import sys
+import click
+import re
+
+
+def parse_expected_transfers(ground_truth_f):
+    """ Parse ALF's log file.
+
+    Parameters
+    ----------
+    ground_truth_f: string
+        file descriptor to ALF's log file
+
+    Returns
+    -------
+    expected_transfers: list of tuples
+        list of transfers with each tuple representing (organism donor, gene
+        donated, organism recipient, gene received)
+
+    Notes
+    -----
+    Report horizontal gene transfers with information on organism donor,
+    gene donated, organism recipient and gene received.
+    """
+    expected_transfers = []
+    string = "lgt from organism "
+    for line in ground_truth_f:
+        if string in line:
```

Comment:
Oh, I see. Thanks!

On Wed, Sep 16, 2015 at 7:59 AM, Evguenia Kopylova <notifications@github.com
> wrote:

> In benchmark/compute_accuracy.py
> <https://github.com/biocore/WGS-HGT/pull/5#discussion_r39632342>:
>
> > +
> > +    Returns
> > +    -------
> > +    expected_transfers: list of tuples
> > +        list of transfers with each tuple representing (organism donor, gene
> > +        donated, organism recipient, gene received)
> > +
> > +    Notes
> > +    -----
> > +    Report horizontal gene transfers with information on organism donor,
> > +    gene donated, organism recipient and gene received.
> > +    """
> > +    expected_transfers = []
> > +    string = "lgt from organism "
> > +    for line in ground_truth_f:
> > +        if string in line:
>
> I'm not sure this will work because the line doesn't start with the string lgt
> from organism, the format of the log file is:
> ..
>
> time 3.2440: lgt from organism 2 with gene 2594 to organism 6, now gene 4126
>     orthologues replacement, gene 2594 in organism 6 deleted
> ...
> where the `time` is variant (you can see an example at the bottom of `test_compute_accuracy.py`)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/WGS-HGT/pull/5/files#r39632342>.
>